src/helpers/xml-builder.js

```diff
@@ -1430,17 +1431,31 @@ const buildTable = (vNode, attributes, docxDocumentInstance) => {
       minimumWidth = Math.round((percentageValue / 100) * attributes.maximumWidth);
     }
 
+    // Calculate maximum width of table
+    if (pixelRegex.test(tableStyles['max-width'])) {
+      pixelRegex.lastIndex = 0;
+      maximumWidth = pixelToTWIP(tableStyles['max-width'].match(pixelRegex)[1]);
+    } else if (percentageRegex.test(tableStyles['max-width'])) {
+      percentageRegex.lastIndex = 0;
+      const percentageValue = tableStyles['max-width'].match(percentageRegex)[1];
+      maximumWidth = Math.round((percentageValue / 100) * attributes.maximumWidth);
+    }
+
     // Calculate specified width of table
     if (pixelRegex.test(tableStyles.width)) {
+      pixelRegex.lastIndex = 0;
       width = pixelToTWIP(tableStyles.width.match(pixelRegex)[1]);
     } else if (percentageRegex.test(tableStyles.width)) {
+      percentageRegex.lastIndex = 0;
       const percentageValue = tableStyles.width.match(percentageRegex)[1];
       width = Math.round((percentageValue / 100) * attributes.maximumWidth);
     }
 
     // Choose the higher of the two widths
     modifiedAttributes.width =
-      width && minimumWidth ? Math.max(width, minimumWidth) : width || minimumWidth;
+      width && minimumWidth
+        ? Math.max(Math.min(width, maximumWidth), minimumWidth)
+        : Math.min(maximumWidth, width || minimumWidth || Number.POSITIVE_INFINITY);
```

Comment:
I elaborated all conditions for clarity, can you please check if the existing code covers all the below.
```suggestion
      if ( width ) {
          modifiedAttributes.width = width;
          if ( minimumWidth ) modifiedAttributes.width = Math.max(modifiedAttributes.width, minimumWidth);
          if ( maximumWidth ) modifiedAttributes.width = Math.min(modifiedAttributes.width, maximumWidth);
      } else {
           if ( minimumWidth  || maximumWidth ) {
              // max overrides min if both are specified
              modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.max(attributes.maximumWidth, minimumWidth);
          }
          
          if ( !minimumWidth  && !maximumWidth ) {
              // neither is specified, let word auto-process it
              delete modifiedAttributes.width;
          }
      }
```

Reply:
@amrita-syn Whats with this line, 

modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.max(attributes.maximumWidth, minimumWidth); ? **Math.min(attributes.maximumWidth, maximumWidth)**

Reply:
`width = Math.min(width, attributes.maximumWidth)` at the end should handle every situation where computed width might be greater than page width.

Reply:
`modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.max(attributes.maximumWidth, minimumWidth);` 
will need to be 
`modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.min(attributes.maximumWidth, minimumWidth);`